### PR TITLE
Switch to better implementation of priming images for goldens

### LINKED ISSUE #36: screenMatchesGolden sometimes creates failure folders even when the test succeeds
This seems to be caused by `await _primeImages(fileName, actualFinder);`.
Sometimes things are not completely settled here and a failure folder is created.
The test ist still successful but failure artifacts are published in the CI system.

I can work around it by using `pumpAndSettle` in the `deviceSetup` but it seems weird.

Why do the golden images needed to be primed at all? Commenting this out doesn't seem to have an effect on duration or test results.

Thanks for the library!

### PR DESCRIPTION
**Pull request by coreysprague**
Fixes #36

### CODE REVIEW

File: packages/golden_toolkit/example/test/example_test.dart [hunk 1]
```diff
@@ -85,6 +85,24 @@ void main() {
         overrideGoldenHeight: 1200,
       );
     });
+
+    group('GoldenBuilder examples of accessibility testing', () {
```
Comment (coreysprague) [inline packages/golden_toolkit/example/test/example_test.dart:89]: re-instated... I accidentally nuked this a couple days ago...

File: packages/golden_toolkit/lib/src/testing_tools.dart [hunk 2]
```diff
@@ -249,7 +252,52 @@ Future<void> compareWithGolden(
...
+  // We are just trying to waste some time that matches the time needed to call matchesGoldenFile.
+  // This should be enough time for most images/assets to be ready.
+  await tester.runAsync<void>(() async {
+    final image = await layer.toImage(renderObject.paintBounds);
+    await image.toByteData(format: ImageByteFormat.png);
+    await image.toByteData(format: ImageByteFormat.png);
+  });
+}
+
+/// A function that waits for all [Image] widgets found in the widget tree to finish decoding.
```
Comment (mZadorskii) [inline packages/golden_toolkit/lib/src/testing_tools.dart:277]: May be some callout that currently supported:
**Images, Decorations**

File: packages/golden_toolkit/lib/src/widget_tester_extensions.dart [hunk 3]
```diff
@@ -3,10 +3,22 @@ import 'dart:ui';
...
+import 'configuration.dart';
 import 'device.dart';
 
+/// Convenience extensions on WidgetTester
+extension WidgetTesterImageLoadingExtensions on WidgetTester {
+  /// Waits for images to decode. Use this to ensure that images are properly displayed
+  /// in Goldens. The implementation of this can be configured as part of GoldenToolkitConfiguration
+  ///
+  /// If you have assets that are not loading with this implementation, please file an issue and we will explore solutions.
+  Future<void> waitForAssets() {
```
Comment (mZadorskii) [inline packages/golden_toolkit/lib/src/widget_tester_extensions.dart:15]: Future<void> waitForAssets() =>  Future<void> waitForAssets() 
:)

**Comment by coreysprague** [comment]
@christian-muertz I merged your PR into here and added the following:
- primed BoxDecoration images
- added tests for the specific priming funcs
- added an extension on WidgetTester to expose the waitForAssets() method
- made your new method the default.

Thanks again for your help!

**Comment by christian-muertz** [comment]
> @christian-muertz I merged your PR into here and added the following:
> 
> - primed BoxDecoration images
> 
> - added tests for the specific priming funcs
> 
> - added an extension on WidgetTester to expose the waitForAssets() method
> 
> - made your new method the default.
> 
> 
> 
> Thanks again for your help!
> 
> 
 
I left a comment at the new impl.

**Comment by coreysprague** [comment]
@christian-muertz I don't see the comment. Can you re-post?

**Comment by christian-muertz** [comment]
Just forgot to finish the review

**Comment by coreysprague** [comment]
@christian-muertz you are correct. I'll throw a quick PR up to resolve. I also updated our repo to use this and found 2 out of about 500 goldens that had issues, so I'm going to dig in to try to figure out the gaps.

**Pull request merged**